index.js

```diff
@@ -1,21 +1,41 @@
 'use strict';
 
 const config = require('./config');
-const amqp = require('amqp-wrapper')(config.amqp);
-const knex = require('knex')(config.db.knex);
 const debug = require('debug')('amqp-log-to-pg');
 
-knex.schema.createTableIfNotExists(config.db.tableName, function (table) {
-  table.increments();
-  table.jsonb('data');
-  table.timestamps(true, true);
-}).then(function () {
-  amqp.connect().then(function () {
-    amqp.consume(onMessage);
-  }).catch(console.error);
-});
+let hasMeta;
+function migrate (knex) {
+  return knex.schema.createTableIfNotExists(config.db.tableName, function (table) {
+    table.increments();
+    table.jsonb('meta');
+    table.jsonb('data');
+    table.timestamps(true, true);
+  }).then(() => {
+    return knex.schema.hasColumn(config.db.tableName, 'meta');
+  });
+}
+
+function main () {
+  const knex = require('knex')(config.db.knex);
+  migrate(knex).then(function (result) {
+    const amqp = require('amqp-wrapper')(config.amqp);
+    hasMeta = result;
+    amqp.connect().then(function () {
+      amqp.consume(onMessage);
+    }).catch(console.error);
+  });
+
+  function onMessage (message, cb) {
+    debug('message received', message);
+    const insert = { data: JSON.stringify(message) };
```

Comment:
should this all be in a try/catch block? incase `JSON.stringify` explodes?

Reply:
Which shouldn't happen in most cases... the only thing I can think of might be circular references.
```js
const x = { a: 1 };
x['x'] = x;
JSON.stringify(x);
// Uncaught TypeError: Converting circular structure to JSON
```